engine/crates/graphql-schema-diff/src/patch/type_definitions.rs

```diff
@@ -0,0 +1,318 @@
+        TypeDefinition::Interface(interface) => interface.implements_interfaces().collect(),
+        _ => Vec::new(),
+    };
+
+    implements.extend(paths.added_interface_impls(ty.name()));
+    implements.retain(|interface| !paths.is_interface_impl_removed(ty.name(), interface));
+    implements.sort_unstable();
+    implements.dedup();
+
+    if !implements.is_empty() {
+        schema.push_str(" implements ");
+        schema.push_str(&implements.join(" & "));
+    }
+
+    patch_directives(ty.directives(), schema, paths);
+
+    match ty {
+        TypeDefinition::Scalar(_) => (),
+        TypeDefinition::Object(object) => patch_fields(object.fields(), ty.name(), schema, paths),
+        TypeDefinition::Interface(interface) => patch_fields(interface.fields(), ty.name(), schema, paths),
+        TypeDefinition::Union(union) => patch_union(union.members(), ty.name(), schema, paths),
+        TypeDefinition::Enum(r#enum) => patch_enum_values(r#enum.values(), ty.name(), schema, paths),
+        TypeDefinition::InputObject(input_object) => {
+            patch_input_object(input_object.fields(), ty.name(), schema, paths)
+        }
+    }
+
+    schema.push_str("\n\n");
+}
+
+fn patch_input_object<'a, T: AsRef<str>>(
+    fields: impl Iterator<Item = InputValueDefinition<'a>>,
+    parent: &str,
+    schema: &mut String,
+    paths: &Paths<'a, T>,
+) {
+    schema.push_str(" {\n");
+
+    let mut removed_fields = Vec::new();
+    let mut changed_field_types = Vec::new();
+
+    for change in paths.iter_second_level(parent) {
+        match change.kind() {
+            ChangeKind::ChangeFieldType => {
+                let field_name = change.second_level().expect("ChangeFieldType without field name");
+                changed_field_types.push((field_name, change.resolved_str()));
+            }
+            ChangeKind::RemoveField => {
+                let field_name = change.second_level().expect("RemoveField without field name");
+                removed_fields.push(field_name);
+            }
+            ChangeKind::AddField => {
+                schema.push_str(INDENTATION);
+                schema.push_str(change.resolved_str().trim());
+                schema.push('\n');
+            }
+            kind => {
+                debug_assert!(false, "Unhandled change at `{path}`: {kind:?}", path = change.path())
+            }
+        }
+    }
+
+    removed_fields.sort_unstable();
+
+    for field in fields {
+        if removed_fields.binary_search(&field.name()).is_ok() {
+            continue;
+        }
+
+        schema.push_str(INDENTATION);
+        schema.push_str(field.name());
+
+        schema.push_str(": ");
+
+        if let Ok(idx) = changed_field_types.binary_search_by(|(field_name, _)| field_name.cmp(&field.name())) {
+            schema.push_str(changed_field_types[idx].1);
+        } else {
+            schema.push_str(&field.ty().to_string());
+        }
+
+        schema.push('\n');
```

Comment:
Are directives & defaults intentionally missing here?  Assume so, just checking

Reply:
Yes I have issues for these.